src/Azure/Orleans.Transactions.AzureStorage/TransactionalState/AzureTableTransactionalStateStorage.cs

```diff
@@ -0,0 +1,184 @@
+        {
+            try
+            {
+                var batchOperation = new TableBatchOperation();
+
+                this.key.ETag = expectedETag;
+                this.key.Metadata = metadata;
+                if (string.IsNullOrEmpty(this.key.ETag))
+                    batchOperation.Insert(this.key);
+                else
+                    batchOperation.Replace(this.key);
+
+                // add new states
+                List<Tuple<string,long>> stored = this.states.Select(s => Tuple.Create(s.TransactionId, s.SequenceId)).ToList();
+                List<StateEntity> newStates = new List<StateEntity>();
+                foreach (PendingTransactionState<TState> pendingState in statesToPrepare.Where(p => !stored.Contains(Tuple.Create(p.TransactionId, p.SequenceId))))
+                {
+                    var newState = StateEntity.Create(this.JsonSettings, this.partition, pendingState);
+                    newStates.Add(newState);
+                    batchOperation.Insert(newState);
+                }
+
+                if (batchOperation.Count > AzureTableConstants.MaxBatchSize)
+                {
+                    this.logger.LogError("Too many pending states. PendingStateCount {PendingStateCount}.", batchOperation.Count);
+                    throw new InvalidOperationException($"Too many pending states. PendingStateCount {batchOperation.Count}");
+                }
+
+                await table.ExecuteBatchAsync(batchOperation).ConfigureAwait(false);
+                this.states.AddRange(newStates);
+                return this.key.ETag;
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogError("Transactional state persist failed {Exception}.", ex);
+                throw;
+            }
+        }
+
+        public async Task<string> Confirm(string expectedETag, string metadata, string transactionIdToCommit)
+        {
+            try
+            {
+                // only update storage if transaction id is greater then previously committed
+                if (string.Compare(transactionIdToCommit, this.key.CommittedTransactionId) <= 0)
+                    return this.key.ETag;
+
+                if (!TryFindState(transactionIdToCommit, out TState state))
+                {
+                    this.logger.LogCritical("Transactional state non-recoverable error.  Attempting to confirm a transaction {TransactionId} for which no state exists.", transactionIdToCommit);
+                    throw new InvalidOperationException($"Transactional state non-recoverable error.  Attempting to confirm a transaction {transactionIdToCommit} for which no state exists.");
+                }
+
+                this.key.ETag = expectedETag;
+                this.key.Metadata = metadata;
+                this.key.CommittedTransactionId = transactionIdToCommit;
+                await WriteKey();
```

Comment:
ConfigureAwait?

Reply:
added it inside write key, where the external call is made.

Reply:
If you have to pick one place to add the call, it should be in the top-level async method. The effect of `ConfigureAwait(false)` is local to the method which it is used in, it has no effect on the caller to that method.

In other words, this `await` here will continue on the captured context

Reply:
bleh, I read up on this more.. ConfigureAwait sux.  Looks like it needs to be called at every level.  I thought it was only needed on the async call that actually blocked.   Will fix.. again.. bleh!??